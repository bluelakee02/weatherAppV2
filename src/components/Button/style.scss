.mainButton {
  //border: 5px solid $green;
  background-color: $blue-01;
  color: #fff;
  border: 0;
  margin-top: 20px;

  &:hover {
    filter: brightness(120%);
  }
  font-size: .875rem;
  line-height: 1.29;
  letter-spacing: .16px;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  max-width: 20rem;
  min-height: 3rem;
  padding: calc(0.875rem - 3px) 63px calc(0.875rem - 3px) 15px;
  text-align: left;
  text-decoration: none;
  outline: none;
  cursor: pointer;
}

.rotate {
  transition: transform .35s ease-in-out;

  &.down {
    transform: rotate(180deg);
  }
}
