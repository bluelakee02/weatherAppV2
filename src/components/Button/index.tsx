import style from './style.scss';
import * as React from 'react';

type ButtonProps = {classname?: string, label: string, onClick?, type?};

const Button = (props: ButtonProps) => (
    <button onClick={props.onClick} className={`${style.mainButton} ${props.classname ?? ''}`}>{props.label}</button>
);

export default Button;
