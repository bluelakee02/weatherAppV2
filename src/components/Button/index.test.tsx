import * as React from 'react';
import {shallow, mount, render} from 'enzyme';

import Button from './index';

describe('Button test', () => {
  it('should match the snapshot', () => {
    expect(shallow(<Button/>)).toMatchSnapshot();
  });

  it('should be selectable by class', () => {
    expect(shallow(<Button/>).is('.mainButton')).toBe(true);
  });

  it('should mount in a full DOM', () => {
    expect(mount(<Button/>).find('.mainButton').length).toBe(1);
  });

  it('should render to static HTML', () => {
    expect(render(<Button label="Hello" />).text()).toEqual('Hello');
  });
});
