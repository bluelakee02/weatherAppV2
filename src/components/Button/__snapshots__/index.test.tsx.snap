// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test should match the snapshot 1`] = `
<button
  className="mainButton "
/>
`;
